HealthcareSystem/Backend/Service/UsersAndWorkingTime/PatientService.cs

```
@@ -14,16 +14,21 @@ namespace Service.UsersAndWorkingTime
 {
     public class PatientService
     {
+        private IActivePatientRepository _activePatientRepository;
+        private IDeletedPatientRepository _deletedPatientRepository;
 
-        private ActivePatientRepository activePatientRepository = new ActivePatientRepository();
-        private DeletedPatientRepository deletedPatientRepository = new DeletedPatientRepository();
+        public PatientService(IActivePatientRepository activePatientRepository,IDeletedPatientRepository deletedPatientRepository)
+        {
+            _activePatientRepository = activePatientRepository;
+            _deletedPatientRepository = deletedPatientRepository;
+        }
         public Patient RegisterPatient(Patient patient)
         {
             if (!patient.IsGuest &&  (!IsUsernameValid(patient.Username) || !IsPasswordValid(patient.Password)))
             {
                 return null;
             }
-            return activePatientRepository.NewPatient(patient);
+            return _activePatientRepository.AddPatient(patient);
         }
```

I would also throw exceptions here instead of return null.